### DISCUSSION TITLE
Lock Coordinate Cursor to loaction option.

### DISCUSSION THREAD

**Discussion by Joku-Suomalainen** [opengisch/QField#4408] [Ideas]
Currently the "Lock coordinate cursor" button is easy to enable/disable and so also easy to disable accidentally. This has been a problem for new qfield users and has caused problems multiple times now. 
Could it be possible to add and option in the options menu to permanently lock the coordinate cursor while the option is on?
This would help a lot.

**Comment by amgiss** [comment]
You can record the @position_source_name in a data field and use constraints to warn the user the cursor is not locked to the current position (Orange Form) or block the recording by checking "Enforce expression constraint" (Red Form). Something like "Your Field Name" <> 'manual' as the constraint expression works for this.

**Reply by DrKraill** [reply]
Hi there, apologies for a potentially silly question (I'm very new to these additional variables) but does it matter which field holds the expression?

For example, I have a point layer with several fields, can I enter the constraint you suggested in the FID? Therefore, "FID <> manual" (with enforce expression constraint checked).

Many thanks for your time

**Comment by DrKraill** [comment]
Hi there, can I ask if you created a solution for this? I'm struggling with adding an enforced constraint to one of the fields in the attribute table.

**Comment by amgiss** [comment]
You would need to use a text field in your table with a default value expression of @position_source_name for this to work. Android devices will return either "android" when the cursor is locked or "manual" when the cursor is unlocked.

**Reply by DrKraill** [reply]
Excellent, understood and many thanks.
